controller/work-item-children-blackbox_test.go

```diff
@@ -0,0 +1,263 @@
+
+	svc = testsupport.ServiceAsUser("WorkItemChildren-Service", almtoken.NewManagerWithPrivateKey(priv), s.testIdentity)
+	require.NotNil(s.T(), svc)
+	s.workItemChildrenCtrl = NewWorkItemChildrenController(svc, s.db)
+	require.NotNil(s.T(), s.workItemChildrenCtrl)
+
+	svc = testsupport.ServiceAsUser("TestWorkItem-Service", almtoken.NewManagerWithPrivateKey(priv), testIdentity)
+	require.NotNil(s.T(), svc)
+	s.svc = svc
+	s.workItemCtrl = NewWorkitemController(svc, s.db)
+	require.NotNil(s.T(), s.workItemCtrl)
+
+	svc = testsupport.ServiceAsUser("Space-Service", almtoken.NewManagerWithPrivateKey(priv), testIdentity)
+	require.NotNil(s.T(), svc)
+	s.spaceCtrl = NewSpaceController(svc, s.db)
+	require.NotNil(s.T(), s.spaceCtrl)
+
+}
+
+// The SetupTest method will be run before every test in the suite.
+// SetupTest ensures that none of the work item links that we will create already exist.
+// It will also make sure that some resources that we rely on do exists.
+func (s *workItemChildSuite) SetupTest() {
+	var err error
+
+	// Create a work item link space
+	createSpacePayload := CreateSpacePayload("test-space"+uuid.NewV4().String(), "description")
+	_, space := test.CreateSpaceCreated(s.T(), s.svc.Context, s.svc, s.spaceCtrl, createSpacePayload)
+	s.userSpaceID = *space.Data.ID
+	s.T().Logf("Created link space with ID: %s\n", *space.Data.ID)
+
+	// Create 3 work items (bug1, bug2, and feature1)
+	bug1Payload := CreateWorkItem(s.userSpaceID, workitem.SystemBug, "bug1")
+	_, bug1 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.workItemCtrl, bug1Payload)
+	require.NotNil(s.T(), bug1)
+
+	s.bug1ID, err = strconv.ParseUint(*bug1.Data.ID, 10, 64)
+	require.Nil(s.T(), err)
+	s.T().Logf("Created bug1 with ID: %s\n", *bug1.Data.ID)
+
+	bug2Payload := CreateWorkItem(s.userSpaceID, workitem.SystemBug, "bug2")
+	_, bug2 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.workItemCtrl, bug2Payload)
+	require.NotNil(s.T(), bug2)
+
+	s.bug2ID, err = strconv.ParseUint(*bug2.Data.ID, 10, 64)
+	require.Nil(s.T(), err)
+	s.T().Logf("Created bug2 with ID: %s\n", *bug2.Data.ID)
+
+	bug3Payload := CreateWorkItem(s.userSpaceID, workitem.SystemBug, "bug3")
+	_, bug3 := test.CreateWorkitemCreated(s.T(), s.svc.Context, s.svc, s.workItemCtrl, bug3Payload)
+	require.NotNil(s.T(), bug3)
+
+	s.bug3ID, err = strconv.ParseUint(*bug3.Data.ID, 10, 64)
+	require.Nil(s.T(), err)
+	s.T().Logf("Created bug3 with ID: %s\n", *bug3.Data.ID)
+
+	// Create a work item link category
+	createLinkCategoryPayload := CreateWorkItemLinkCategory("test-user" + uuid.NewV4().String())
+	_, workItemLinkCategory := test.CreateWorkItemLinkCategoryCreated(s.T(), s.svc.Context, s.svc, s.workItemLinkCategoryCtrl, createLinkCategoryPayload)
+	require.NotNil(s.T(), workItemLinkCategory)
+	s.userLinkCategoryID = *workItemLinkCategory.Data.ID
+	s.T().Logf("Created link category with ID: %s\n", *workItemLinkCategory.Data.ID)
+
+	s.DB.Unscoped().Delete(&link.WorkItemLinkType{})
```

Comment:
@baijum @aslakknutsen for a second I though I might have missed this delete. But we had discussed it and found a way around it.